### PR TITLE
[fix](catalog)add custom jar

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

- allow put custom jar in `${DORIS_HOME}/lib/java_extensions/custom_extension` such as `paimon-s3-0.4.0-incubating.jar`
- add some note for paimon and fqdn

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: bin/start_be.sh [hunk 1]
```diff
@@ -90,6 +90,15 @@ for preload_jar in "${preload_jars[@]}"; do
     fi
 done
 
+# add custom jar
+custom_extension_folder="${DORIS_HOME}/lib/java_extensions/custom_extension"
```
Comment (wsjz) [inline bin/start_be.sh:94]: use build-in preload-extensions

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Review by BePPPower — APPROVED** [review]

**Review by kaka11chen — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.81 seconds
 stream load tsv:          541 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17161976619 Bytes

**Pull request merged**